File: DSA_CSHARP/Recursion/Factorial.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA_CSHARP.Recursion;
internal class Factorial
{
    public int FactorialWithN(int n)
    {

        if (n > 0)
            return FactorialWithN(n - 1) * n;
        return 1;
    }
    public string Main()
    {
        Console.WriteLine($"------------------------------------");
        Console.Write($"Enter 'N': ");

        var n = Convert.ToInt32(Console.ReadLine());

        var res = FactorialWithN(n);

        Console.WriteLine($"Factorial of {n} is: {res}");
        return res.ToString();
    }
}

File: DSA_PYTHON/Recursion/Factorial.py
class Factorial:
    def factorial(self,n):
        if(n>0):
            return self.factorial(n-1)*n
        return 1

    def main(self):
        print("------------------------------------")
        print("Enter 'N'")
    
        n = int(input())
        res=self.factorial(n)
    
        print(f"Factorial of {n} is: {res}")
        return res
